Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -36,11 +37,11 @@ internal class HttpCoreDiagnosticSourceListener : IObserver<KeyValuePair<string,
         private readonly PropertyFetcher deprecatedResponseFetcher = new PropertyFetcher("Response");
         private readonly PropertyFetcher deprecatedRequestGuidFetcher = new PropertyFetcher("LoggingRequestId");
         private readonly PropertyFetcher deprecatedResponseGuidFetcher = new PropertyFetcher("LoggingRequestId");
-        
+
         #endregion
 
-        private readonly ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>> pendingTelemetry =
-            new ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>>();
+        private readonly ConditionalWeakTable<HttpRequestMessage, IOperationHolder<DependencyTelemetry>> pendingTelemetry = 
```

Comment:
I was thinking to call Begin callback many times and check somehow that the objects will be gone after `GC.Collect()`. However I cannot think of easy way to check that those operation holders are disposed. 

BTW, will dispose be called on `IOperationHolder`? Will it send telemetry on dispose?